### ISSUE TITLE
Implement SoftSpokenOT

### ISSUE THREAD

**Issue by sinui0** [privacy-ethereum/mpz#41]
With the security of KOS called into question #10 , we should deprecate it in favor of an implementation of [SoftSpokenOT](https://eprint.iacr.org/2022/192).

Reference impl available from [libOTe](https://github.com/osu-crypto/libOTe/tree/ad07f27b9ab229fa1d64ee642798c16ea5d8813f/libOTe/TwoChooseOne/SoftSpokenOT).

**Comment by matsutakk** [comment]
I will try to work on this.

**Comment by cheng-wei-huang0612** [comment]
Can we take this task? @sinui0

**Comment by sinui0** [comment]
> Can we take this task? @sinui0

Yes, it is still open for contributors. Please do refer to our [standards](https://github.com/privacy-scaling-explorations/mpz/blob/65994b094d28bac6bbf53be35e6e3002e4fd2573/DESIGN.md) doc beforehand :pray: 

I would suggest first implementing the core functionality without any IO.

**Comment by mahmudsudo** [comment]
can i take on this ?

**Comment by mahmudsudo** [comment]
@sinui0 can i take this?

**Comment by sinui0** [comment]
> [@sinui0](https://github.com/sinui0) can i take this?

I expect this to be a difficult task. If you feel you can tackle it and provide an impl at a high quality standard then we would of course gladly accept it 👍

**Comment by sinui0** [comment]
Closed by #433
